BlazarData/src/main/java/com/hubspot/blazar/data/dao/StateDao.java

```diff
@@ -44,21 +42,45 @@
 
   @SingleValueResult
   @SqlQuery("" +
-      "SELECT module.*, " +
-      "     lastSuccessfulBuild.*, " +
-      "     lastNonSkippedBuild.*, " +
-      "     lastBuild.*, " +
-      "     inProgressBuild.*, " +
-      "     pendingBuild.* " +
-      "  FROM modules AS module " +
-      "     LEFT OUTER JOIN module_builds AS lastSuccessfulBuild ON (module.id = lastSuccessfulBuild.moduleId) " +
-      "     LEFT OUTER JOIN module_builds AS lastNonSkippedBuild ON (module.id = lastNonSkippedBuild.moduleId) " +
-      "     LEFT OUTER JOIN module_builds AS lastBuild ON (module.lastBuildId = lastBuild.id) " +
-      "     LEFT OUTER JOIN module_builds AS inProgressBuild ON (module.inProgressBuildId = inProgressBuild.id) " +
-      "     LEFT OUTER JOIN module_builds AS pendingBuild ON (module.pendingBuildId = pendingBuild.id) " +
-      "  WHERE module.id = :id " +
-      "  AND lastNonSkippedBuild.state != 'SKIPPED' " +
-      "  AND lastSuccessfulBuild.state = 'SUCCEEDED' " +
-      "  ORDER BY lastNonSkippedBuild.id DESC, lastSuccessfulBuild.id DESC LIMIT 1")
-  Optional<ModuleState> getModuleStateById(@BindWithRosetta Module module);
+     "SELECT repositoryBuild.*, lastSuccessfulBuild.* "+
+      "FROM module_builds AS lastSuccessfulBuild " +
+      "LEFT OUTER JOIN repo_builds AS repositoryBuild on (lastSuccessfulBuild.repoBuildId = repositoryBuild.id) " +
+      "WHERE lastSuccessfulBuild.moduleId = :moduleId " +
+      "AND lastSuccessfulBuild.state = 'SUCCEEDED' " +
+      "ORDER BY lastSuccessfulBuild.id DESC LIMIT 1")
+  Optional<BuildInfo> getLastSuccessfulBuildInfo(@Bind("moduleId") int moduleId);
+
+  @SingleValueResult
+  @SqlQuery("" +
+      "SELECT repositoryBuild.*, lastNonSkippedBuild.* " +
+      "FROM module_builds AS lastNonSkippedBuild " +
+      "LEFT OUTER JOIN repo_builds AS repositoryBuild on (lastNonSkippedBuild.repoBuildId = repositoryBuild.id) " +
+      "WHERE lastNonSkippedBuild.moduleId = :moduleId " +
+      "AND lastNonSkippedBuild.state in ('QUEUED', 'WAITING_FOR_UPSTREAM_BUILD', 'LAUNCHING', 'IN_PROGRESS', 'SUCCEEDED', 'CANCELLED', 'FAILED') " +
```

Comment:
Potential tweak here would be to have a subquery here to get the possible states and pull out skipped so we don't have to update this query in future.